recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,176 @@
+                    "mesa-utils-extra",
+                    "libgl1-mesa-dev",
+                    "libglapi-mesa",
+                    "libsm-dev",
+                    "libx11-dev",
+                    "libxext-dev",
+                    "libxt-dev",
+                    "libglu1-mesa-dev"]
+
+        if pack_names:
+            installer = tools.SystemPackageTool()
+            for item in pack_names:
+                installer.install(item + self._system_package_architecture())
+
+    def config_options(self):
+        if self.settings.compiler == "Visual Studio":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_TESTING"] = "OFF"
+        cmake.definitions["BUILD_EXAMPLES"] = "OFF"
+        cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+
+        if self.options.minimal:
+            cmake.definitions["VTK_Group_StandAlone"] = "OFF"
+            cmake.definitions["VTK_Group_Rendering"] = "OFF"
+        if self.options.ioxml:
+            cmake.definitions["Module_vtkIOXML"] = "ON"
+        if self.options.ioexport:
+            cmake.definitions["Module_vtkIOExport"] = "ON"
+        if self.options.qt:
+            cmake.definitions["VTK_Group_Qt"] = "ON"
+            cmake.definitions["VTK_QT_VERSION"] = "5"
+            cmake.definitions["VTK_BUILD_QT_DESIGNER_PLUGIN"] = "OFF"
+        if self.options.mpi:
+            cmake.definitions["VTK_Group_MPI"] = "ON"
+            cmake.definitions["Module_vtkIOParallelXML"] = "ON"
+        if self.options.mpi_minimal:
+            cmake.definitions["Module_vtkIOParallelXML"] = "ON"
+            cmake.definitions["Module_vtkParallelMPI"] = "ON"
+
+        if self.settings.os == 'Macos':
+            self.env['DYLD_LIBRARY_PATH'] = os.path.join(self.build_folder, 'lib')
+            self.output.info("cmake build: %s" % self.build_folder)
+
+        cmake.configure(build_folder='build')
+        return cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        # "$\package\lib\vtk" contains "hierarchy\conanvtk\" and a lot of *.txt files in it.
+        shutil.rmtree(os.path.join(self.package_folder, 'lib', 'vtk')) #
+        # "$\package\lib\cmake" contains a lot of *.cmake files. conan-center HOOK disallow *.cmake files in package.
+        shutil.rmtree(os.path.join(self.package_folder, 'lib', 'cmake'))
+        # Licences are created in "$\package\share\licenses\conanvtk\" while their must be in "$\package\licenses\"
+        os.rename(os.path.join(self.package_folder, 'share', 'licenses', 'conanvtk'), os.path.join(self.package_folder, 'licenses'))
+        shutil.rmtree(os.path.join(self.package_folder, 'share'))
```

Comment:
You can use `tools.rmdir`.
I did not check, but you might be able to remove `import shutil` from the top.
```suggestion
        # "$\package\lib\vtk" contains "hierarchy\conanvtk\" and a lot of *.txt files in it.
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'vtk'))
        # "$\package\lib\cmake" contains a lot of *.cmake files. conan-center HOOK disallow *.cmake files in package.
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
        # Licences are created in "$\package\share\licenses\conanvtk\" while their must be in "$\package\licenses\"
        tools.rmdir(os.path.join(self.package_folder, 'share', 'licenses', 'conanvtk'), os.path.join(self.package_folder, 'licenses'))
        tools.rmdir(os.path.join(self.package_folder, 'share'))
```

Reply:
Done